retrospective-update/prepare.py

```
@@ -0,0 +1,102 @@
+        if os.path.basename(state) != expected_final_state_file:
+            os.remove(state)
+    # now check if every vpu has the state of interest
+    states = natsorted(glob(os.path.join(FINAL_STATES_DIR, '*', 'finalstate*.parquet')))
+    if len(states) == len(list(glob(os.path.join(CONFIGS_DIR, '*')))):
+        return
+
+    # download the states we need
+    cmd = f's5cmd --no-sign-request cp "{S3_FINAL_STATES_DIR}/*/{expected_final_state_file}" {FINAL_STATES_DIR}/'
+    result = subprocess.run(cmd, shell=True, capture_output=True, text=True, )
+    if result.returncode != 0:
+        cl.ping('FAIL', f"Error running s5cmd copy command")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)